src/onegov/search/mixins.py

```diff
@@ -134,7 +152,129 @@ def es_tags(self):
+        """
+        Drops the full text search column
+
+        :param session: db session
+        :param schema: schema the full text column shall be added
+        :param model: model to drop the index from
+        :param col_name: column name of fts index
+        :return: None
+        """
+        query = f"""
+            ALTER TABLE "{schema}".{model.__tablename__} DROP COLUMN IF EXISTS
+            {col_name};
+        """
+        session.execute(query)
+        session.execute("COMMIT")
+
+    @staticmethod
+    def add_fts_column(session, schema, model, col_name='fts_idx'):
+
+        """
+        This function is used for re-indexing and as migration step moving to
+        postgresql full text search (fts), OGC-508.
+
+        It adds a separate column for the tsvector to `schema`.`table`
+        creating a multilingual gin index on the columns/data defined per
+        model.
+
+        :param session: session
+        :param schema: schema name
+        :param model: model to add the index
+        :param col_name: column name of fts index
+        :return: None
+        """
+
+        # we build the tsvector expression for all supported languages based
+        # on this format
+        # to_tsvector('simple', '<tsvector_string_of_model') ||
+        # to_tsvector('german', '<tsvector_string_of_model') ||
+        # to_tsvector('french', '<tsvector_string_of_model') ||
+        # ...
+
+        tsvector_expression = \
+            Searchable.multi_language_tsvector_expression(
+                model.psql_tsvector_string(model))
+
+        query = f"""
+            ALTER TABLE "{schema}".{model.__tablename__}
+            ADD COLUMN IF NOT EXISTS {col_name} tsvector GENERATED ALWAYS AS
+            ({tsvector_expression}) STORED;
+        """
+        session.execute(query)
+        session.execute("COMMIT")
+
+    @staticmethod
+    def multi_language_tsvector_expression(tsvector_string):
+        return ' || '.join(
+            "to_tsvector('{}', {})".format(language, tsvector_string)
+            for language in get_fts_index_languages()
+        )
+
+    @staticmethod
+    def create_tsvector_string(*cols):
+        """
+        Creates tsvector string for columns
+        Doc reference:
+        https://www.postgresql.org/docs/current/textsearch-tables.html#TEXTSEARCH-TABLES-INDEX
+
+        :param cols: column names to be indexed
+        :return: tsvector string for multiple columns i.e. for columns title
+        and body: coalesce(\"title\", '') || ' ' || coalesce(\"body\", '')
+        Note that the '\"{}\"' escapes the column name in case it is a
+        keyword, see
+        http://www.postgresql.org/docs/current/static/sql-keywords-appendix.html
+        """
+
+        base = "'func.coalesce({}, '')'"
+        ext = " || ' ' || 'func.coalesce({}, '')'"
+
+        exp = base
+        for _ in range(len(cols) - 1):
+            exp += ext
+
+        return exp.format(*cols)
```

Comment:
Ach so, jetzt verstehe ich was du meintest!! Danke

Reply:
After switching everything to statements using func.concat and func.coalesce I ran into 'generation expression is not immutable' error. Searching for this issue brought this page https://stackoverflow.com/questions/60249404/why-am-i-getting-a-an-error-when-creating-a-generated-column-in-postgresql saying concat is not immutable. On the following page
https://stackoverflow.com/questions/12310986/combine-two-columns-and-add-into-one-new-column/12320369#12320369
I found I should use the concat operator ' || ' or writing an own immuatable concat function. Would you also prefer the ' || ' operator?

Reply:
Alright I think you can force SQLAlchemy to use `||` (which is only valid for strings) by using the `concat` method on the expr instead, since the expression should know its value is a string, otherwise you may need to insert a cast to string.
```python
expr = expr.concat(' ').concat(func.coalesce(col, ''))
```
I would recommend searching in the SQLAlchemy docs rather than on stackoverflow, they're quite good.

Reply:
Otherwise there should also be a `op.concat` somewhere